normandy/control/static/control/js/components/RecipeForm.jsx

```diff
@@ -49,14 +53,18 @@ export class RecipeForm extends React.Component {
     })
     .then(response => {
       if (recipeId) {
-        dispatch(ControlActions.makeApiRequest('updateRecipe', {
-          recipe: combinedFormValues,
-          recipeId: recipeId
-        }));
+        submitAction = 'updateRecipe';
+        successAction = (response) => dispatch(recipeUpdated(response));
       } else {
-        dispatch(ControlActions.makeApiRequest('addRecipe', combinedFormValues));
-      }
-    })
+        submitAction = 'addRecipe';
+        successAction = (response) => {
+          dispatch(recipeAdded(response));
+          dispatch(push(`/control/recipe/${response.id}/`));
+        };
+      };
+
+      return dispatch(makeApiRequest(submitAction, requestBody)).then(response => successAction(response));
```

Comment:
I actually think setting up the variables makes it harder to see what this method is doing vs the old method of calling dispatch in two different spots. It's a bit more typing but less state for the programmer to store in their head when reading this.